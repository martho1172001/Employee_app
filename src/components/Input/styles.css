input::placeholder {
    color:  black;
    opacity: 1;
    font-weight: bold;

}

input:focus::placeholder {
    color: rgb(179, 179, 179);
  
}
input:focus {
    border:2px solid rgb(170, 170, 170);

    outline-style: none;
  
    caret-color: rgb(64, 189, 238);
  
}
input{
    height: 50px;
    border: 1px solid #B4B7BD;
border-radius: 5px;
opacity: 1;

}
.input-div{
    width: 80%;
display: flex;
flex-direction: column;
margin: 10px;

}
