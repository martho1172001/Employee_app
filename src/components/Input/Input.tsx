import React from 'react';
import './styles.css';
type InputPropType = {
  label: string|null;
  type: 'text' | 'password' | 'number' | 'date';
  value: string|number;
  onChange: (e) => void;
  placeholder: string;
};

const Input: React.FC<InputPropType> = (props) => {
  return (
    <>
      <div className='input-div'>
        {props.label && <label>{props.label}</label>}
        <input
          type={props.type}
          value={props.value}
          onChange={(e) => props.onChange(e.target.value)}
          placeholder={props.placeholder}
        />
      </div>
    </>
  );
};

export default Input;
