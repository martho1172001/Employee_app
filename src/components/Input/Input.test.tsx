import Input, { InputPropType } from './Input';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Button component testing', () => {
  test('if label rendered correctly', () => {
    const InputProps: InputPropType = {
      label: 'button here',
      value: 'button',
      type: 'text',
      placeholder: 'button placeholder'
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId('input-label-test');

    expect(element.innerHTML).toBe('button here');
  });
  test('if input rendered correctly', () => {
    const InputProps: InputPropType = {
      label: 'button here',
      value: 'button',
      type: 'text',
      placeholder: 'button placeholder'
    };

    render(<Input {...InputProps} />);

    const element = screen.getByTestId('input-input-test');

    expect(element).toMatchSnapshot();
    expect(element.getAttribute('value')).toBe('button');
  });
  test('if on focus label renders correctly', () => {
    const InputProps: InputPropType = {
      label: 'button here',
      value: 'button',
      type: 'text',
      placeholder: 'button placeholder'
    };

    render(<Input {...InputProps} />);

    const element = screen.getByTestId('input-input-test');
    const element_label = screen.getByTestId('input-label-test');

    fireEvent.focus(element);
    expect(element_label.getAttribute('class')).toBe('label focused');
  });
  test('if on blur label renders correctly', () => {
    const InputProps: InputPropType = {
      label: 'button here',
      value: 'button',
      type: 'text',
      placeholder: 'button placeholder'
    };

    render(<Input {...InputProps} />);

    const element = screen.getByTestId('input-input-test');
    const element_label = screen.getByTestId('input-label-test');

    fireEvent.focus(element);
    fireEvent.blur(element);
    expect(element_label.getAttribute('class')).toBe('label');
  });
  test('if onchange works correctly', () => {
    const onChangeMock = jest.fn();
    const InputProps: InputPropType = {
      label: 'button here',
      value: 'button',
      type: 'text',
      placeholder: 'button placeholder',
      onChange: onChangeMock
    };

    render(<Input {...InputProps} />);

    const element = screen.getByTestId('input-input-test');

    fireEvent.change(element);

    expect(element).toMatchSnapshot();
    expect(onChangeMock).toHaveBeenCalled;

  });
  test('if onchange with value works correctly', () => {
    const onChangeMock = jest.fn();
    const InputProps: InputPropType = {
      label: 'button here',
      value: 'button',
      type: 'text',
      placeholder: 'button placeholder',
      onChange: onChangeMock
    };

    render(<Input {...InputProps} />);

    const element = screen.getByTestId('input-input-test');

    fireEvent.change(element, { target: { value: 'New value' } });

    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith('New value');
  });
});
