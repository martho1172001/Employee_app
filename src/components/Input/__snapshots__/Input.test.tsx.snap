// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component testing if input rendered correctly 1`] = `
<input
  data-testid="input-input-test"
  placeholder="button placeholder"
  type="text"
  value="button"
/>
`;

exports[`Button component testing if onchange works correctly 1`] = `
<input
  data-testid="input-input-test"
  placeholder="button placeholder"
  type="text"
  value="button"
/>
`;
